### PR TITLE
Feat/use array modify by

### LINKED ISSUE #42: Feature idea: `useArray` and modify by element id
Hi!
I'm using `useArray` and I'm ok with it but I'd like to be able to modify an element by id and have a re-render.

I've tested it locally and this works for me:

```ts
	const modifyElementById = useCallback(
		(id: string, newValue: Partial<T>) =>
			setValue(arr => arr.map(v => (v.id === id ? { ...v, ...newValue } : v))),
		[],
	);
```

I'd like to write a PR (it is a super simple idea) but I open this issue to first ask what do you think about it.

### PR DESCRIPTION
**Pull request by nkint**
As discussed, close #42 , let me know if you require any further edits.
Thanks, great project!

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -192,6 +192,18 @@ describe('useArray', () => {
...
 
+  it('should modify item by id', () => {
+    const { result } = renderHook(() => useArray([{ id: 1, foo: true }, { id: 2, foo: false }]));
+    const { modifyById } = result.current;
+    expect(result.current.value.length).toBe(2);
+
+    act(() => modifyById(2, { foo: true }));
+
+    expect(
+      result.current.value.filter((element: { id: number; foo: boolean }) => element.id === 2)[0].foo,
```
Comment (RIP21) [inline src/index.test.ts]: Use `find` function instead, and not inline pls.
```suggestion
      modifiedElement.foo,
```

File: README-ARRAY.md [hunk 2]
```diff
@@ -112,6 +112,8 @@ Actions:
 - `removeIndex`
 - `removeById` - if array consists of objects with some specific `id` that you pass
 all of them will be removed
+- `modifyById` - if array consists of objects with some specific `id` that you pass
+the element will be modified
```
Comment (RIP21) [inline README-ARRAY.md]:
```suggestion
these elements will be modified.
```

File: src/index.test.ts [hunk 3]
```diff
@@ -192,6 +192,18 @@ describe('useArray', () => {
     expect(result.current.value.length).toBe(1);
   });
 
+  it('should modify item by id', () => {
+    const { result } = renderHook(() => useArray([{ id: 1, foo: true }, { id: 2, foo: false }]));
+    const { modifyById } = result.current;
+    expect(result.current.value.length).toBe(2);
+
+    act(() => modifyById(2, { foo: true }));
+
```
Comment (RIP21) [inline src/index.test.ts]:
```suggestion
const modifiedElement = result.current.value.find((element: { id: number; foo: boolean }) => element.id === 2)
```

**Comment by nkint** [comment]
Hi @RIP21 ! Thanks for the comments.

I understand `modifyByKey` but if we implement this would be nice to be consistent and add also `modifyByKey`.

I think that `id` is a quite common field, maybe we can leave both `removeById`/`modifyById` and `modifyByKey`/`modifyByKey` ?

File: src/index.test.ts (hunk 3 above)
Comment (nkint) [reply src/index.test.ts]: 👍

**Comment by kitze** [comment]
@nkint yes please, add the ones for `id` because 99% of the cases will be by `id`. You can still add the other 2 `byKey` functions.

**Comment by RIP21** [comment]
@nkint merged. Not deployed yet.

**Pull request merged**